Model/Sync/Catalog/Processor.php

```diff
@@ -817,4 +851,42 @@ private function updateSyncTable($syncDataRecord)
             [$syncDataRecord]
         );
     }
+
+    /**
+     * @param string $parentId
+     * @return string
+     */
+    private function ensureEntityExistenceAsProductInYotpo($parentId)
+    {
+        $parentProductsData = $this->getCollectionForSync([$parentId])->getItems();
+        if (!count($parentProductsData)) {
+            return null;
+        }
+
+        $apiUrl = $this->coreConfig->getEndpoint('products');
+        $method = $this->coreConfig->getProductSyncMethod('createProduct');
+        $apiParam = [
+            'url' => $apiUrl,
+            'method' => $method,
+        ];
+
```

Comment:
Hi @EranAvidor / @LiorGingi 

Here we are not dealing 404/409 cases. If you can call getApiParams() (instead of directly calling $apiUrl, $method there ) by passing the productId to this method with other values are blank (or something like that ), this can be manged I think.

Reply:
@manaftryzens can you please share a PR on top of `master` that fixes it based on your understanding and familiarity of the code?

Reply:
@EranAvidor ,

I need these code changes to work on top of that. Shall I fork COR-catalog-fixes branch and create a PR against this COR-catalog-fixes branch?

Reply:
@manaftryzens feel free to fork it and use it as a baseline, please open your PR against `master`.

Reply:
@EranAvidor ,

Please check the below PR https://github.com/YotpoLtd/magento2-module-core/pull/143